### DISCUSSION TITLE
[BUG?] shipping destination - min and max delivery days

### DISCUSSION THREAD

**Discussion by baahar** [etsy/open-api#684] [Q&A]
Hi, 
shipping_profile_destination's return "null" for the attributes **min_** and **max_delivery_days** when there is a shipping carrier selected. The days are returned fine when there is no shipping carrier. 

Can this be fixed or is there another access point where we can get the delivery days for various shipping carriers and mail classes?

Thanks in advance.

**Comment by etsyachristensen** [comment]
@baahar Can you give me an example shipping profile id I can look up to get the destinations and confirm this?

**Reply by baahar** [reply]
Thanks for getting back. An example shipping profile id would be: 17000251

**Reply by etsyachristensen** [reply]
Hmm. I'm looking at that shop in our system but I don't see any apps attached to the shop. Can you tell me what the app name is?

**Reply by baahar** [reply]
It is an old app still on version 2 called ShopShaper. I asked the developer team and they told me how to go about migrating to version 3. I'm still working on it and came across this issue.

**Reply by etsyachristensen** [reply]
Ahh. ok. Unfortunately we can't answer any questions on V2 on this Github board. You'd have to ask in the Google group for V2. You'll need an app on V3 using OAuth 2. Once you have that, if you're still having issues with getting the information from the API, please let us know.

**Reply by baahar** [reply]
My question is about v3 though. I am getting the data from this endpoint:
https://openapi.etsy.com/v3/application/shops/{shop_id}/shipping-profiles/{shipping_profile_id}

{"shipping_profile_id":17000251,"title":"heavy","user_id":-------,
"min_processing_days":1,"max_processing_days":2,
"processing_days_display_label":"1-2 business days","origin_country_iso":"DE",
"origin_postal_code":"64283","profile_type":"manual","is_deleted":false,"domestic_handling_fee":0,
"international_handling_fee":0,"shipping_profile_destinations":[{"shipping_profile_destination_id":10135313826,"shipping_profile_id":17000251,"origin_country_iso":"DE","destination_country_iso":"","destination_region":"none","primary_cost":{"amount":950,"divisor":100,"currency_code":"USD"},"secondary_cost":{"amount":0,"divisor":100,"currency_code":"USD"},"shipping_carrier_id":12,"mail_class":"intl_standard_parcel","min_delivery_days":**null**,"max_delivery_days":**null**},
...

**Reply by etsyachristensen** [reply]
Ok I'm confused. You have a V3 app then? If so, what is the last 6 characters of your API Key? The issue we're having is that we don't see any apps authorized on your shop. Until we can see the shop and the app being used, it's virtually impossible to replicate the issue.

**Reply by baahar** [reply]
I registered the app back when v2 was active. Now I'm migrating and updating my code to only use v3 endpoints. Do I have to contact the team via the Google group in this scenario?

Unfortunately, I don't know why the app is not showing up. The last six characters of my API key are: "zuqbsl"

**Reply by etsyachristensen** [reply]
Thanks. Have you gone through the OAuth consent process with that shop and app? I'll keep trying to find a way to duplicate this.

**Reply by baahar** [reply]
No, I went through the process described in the documentation as "[exchange oauth token 1.0 for oauth token 2.0](https://developers.etsy.com/documentation/essentials/authentication#exchange-oauth-10-token-for-oauth-20-token)".

I will update my code to enable the regular consent process and let you know after that. Thanks for taking the time today.

**Reply by etsyachristensen** [reply]
Ah ok gotcha. yeah it's weird that the app isn't showing up for that shop. Will continue debugging on Monday.
